### PR TITLE
timed job log fixes

### PR DESCRIPTION
**Pull request by mikethebeer**
* added documentation
* fixed a bug in `operations_log_expiration` setting

### CODE REVIEW

File: blackbox/docs/configuration.txt [hunk 1]
```diff
@@ -694,42 +694,59 @@ Collecting Stats
...
 **stats.jobs_log_expiration**
   | *Default:*  ``0s`` (disabled)
   | *Runtime:*  ``yes``
 
-  This setting specifies the period when jobs are stored in the ``sys.jobs_log``
-  table. It periodically clears entries that are completed and older than the
-  specified time interval. This setting overrides ``jobs_log_size`` and logs a
-  warning message if both settings are set explicitly. If the value is set to
-  ``0`` the setting gets disabled.
+  This setting specifies the time period when jobs are cleared out of the
```
Comment (chaudum) [inline blackbox/docs/configuration.txt]: This setting specifies the time period after which jobs are removed from the ... table.

Comment (chaudum) [inline blackbox/docs/configuration.txt]: Entries are cleared by a periodically job when they are older than the specified expire time.

File: blackbox/docs/configuration.txt [hunk 2]
```diff
@@ -694,42 +694,59 @@ Collecting Stats
...
+  :ref:`sys.jobs_log <sys-logs>` table. It periodically clears entries that are
+  completed and older than the specified time interval. This setting overrides
+  :ref:`stats.jobs_log_size <stats.jobs_log_size>`. If the value is set to ``0``
+  the setting gets disabled.
```
Comment (chaudum) [inline blackbox/docs/configuration.txt]: ... the time based log entry eviction is disabled.

File: blackbox/docs/configuration.txt [hunk 3]
```diff
@@ -694,42 +694,59 @@ Collecting Stats
...
+  number results in more expressive results but also in more occupied RAM.
+  Setting it to ``0`` disables collecting operation information.
+
+.. _stats.operations_log_expiration:
+
+**stats.operations_log_expiration**
+  | *Default:*  ``0s`` (disabled)
+  | *Runtime:*  ``yes``
+
+  This setting specifies the time period when operations are cleared out of the
```
Comment (chaudum) [inline blackbox/docs/configuration.txt]: see comments from stats.jobs_log_expiration

File: blackbox/docs/configuration.txt [hunk 4]
```diff
@@ -694,42 +694,59 @@ Collecting Stats
...
-  The number of jobs kept in the ``sys.jobs_log`` table on each node
-  for performance analytics. Older entries will be deleted when the
-  ``jobs_log_size`` is reached. A single SQL statement results in a
-  job to be executed on the cluster. A higher number results in more
-  expressive results but also in more occupied RAM. Setting it to
-  ``0`` disables collecting job information.
+  The number of jobs kept in the :ref:`sys.jobs_log <sys-logs>` table on each
+  node for performance analytics. Older entries will be deleted when the size is
+  reached. A single SQL statement results in a job to be executed on the
+  cluster. A higher number results in more expressive results but also in more
```
Comment (nslater) [inline blackbox/docs/configuration.txt]: "also in more" -> "also more"

File: blackbox/docs/configuration.txt [hunk 5]
```diff
@@ -694,42 +694,59 @@ Collecting Stats
...
 
-  This setting specifies the period when jobs are stored in the ``sys.jobs_log``
-  table. It periodically clears entries that are completed and older than the
-  specified time interval. This setting overrides ``jobs_log_size`` and logs a
-  warning message if both settings are set explicitly. If the value is set to
-  ``0`` the setting gets disabled.
+  This setting specifies the time period when jobs are cleared out of the
+  :ref:`sys.jobs_log <sys-logs>` table. It periodically clears entries that are
+  completed and older than the specified time interval. This setting overrides
+  :ref:`stats.jobs_log_size <stats.jobs_log_size>`. If the value is set to ``0``
```
Comment (nslater) [inline blackbox/docs/configuration.txt]: add comma after ``0``

File: blackbox/docs/configuration.txt [hunk 6]
```diff
@@ -694,42 +694,59 @@ Collecting Stats
...
 
-  The number of operations to keep in the ``sys.operations_log`` table
-  on each node for performance analytics. Older entries will be
-  deleted when the ``operations_log_size`` is reached. A job consists
-  of one or many operations. A higher number results in more
-  expressive results but also in more occupied RAM. Setting it to
-  ``0`` disables collecting operation information.
+  The number of operations to keep in the :ref:`sys.operations_log <sys-logs>`
+  table on each node for performance analytics. Older entries will be deleted
+  when the size is reached. A job consists of one or many operations. A higher
```
Comment (nslater) [inline blackbox/docs/configuration.txt]: "when the size" -> "when this is" (correct?)

File: blackbox/docs/configuration.txt [hunk 7]
```diff
@@ -694,42 +694,59 @@ Collecting Stats
...
+  number results in more expressive results but also in more occupied RAM.
```
Comment (nslater) [inline blackbox/docs/configuration.txt]: "but also in" -> "but also"

File: blackbox/docs/configuration.txt [hunk 8]
```diff
@@ -694,42 +694,59 @@ Collecting Stats
...
+
+**stats.operations_log_expiration**
+  | *Default:*  ``0s`` (disabled)
+  | *Runtime:*  ``yes``
+
+  This setting specifies the time period when operations are cleared out of the
+  :ref:`sys.operations_log <sys-logs>` table. It periodically clears entries
+  that are completed and older than the specified time interval. This setting
+  overrides :ref:`stats.operations_log_size <stats.operations_log_size>`. If the
+  value is set to ``0`` the setting gets disabled.
```
Comment (nslater) [inline blackbox/docs/configuration.txt]: add comma after ``0``

File: blackbox/docs/configuration.txt [hunk 9]
```diff
@@ -752,15 +769,29 @@ Collecting Stats
...
+.. _stats.breaker.jobs_logs.limit:
 
-**stats.breaker.logs.limit**
+**stats.breaker.jobs_log.limit**
   | *Default:*    ``5%``
   | *Runtime:*   ``yes``
 
-  The logs circuit breaker allows CrateDB to prevent job logs data structures
-  from exceeding a certain amount of memory by removing old job logs entries.
+  The jobs log circuit breaker prevents that :ref:`sys.jobs_log <sys-logs>`
```
Comment (nslater) [inline blackbox/docs/configuration.txt]: drop "that"

File: blackbox/docs/configuration.txt [hunk 10]
```diff
@@ -752,15 +769,29 @@ Collecting Stats
...
+  exceeds a certain amount of memory by removing old log entries. The value
```
Comment (nslater) [inline blackbox/docs/configuration.txt]: "exceeds" -> "from exceeding"

"a certain amount of memory" -> "the specified maximum memory"

File: blackbox/docs/configuration.txt [hunk 11]
```diff
@@ -752,15 +769,29 @@ Collecting Stats
...
+  specifies the maximum of memory that can be used out of :ref:`crate-heap-size`.
```
Comment (nslater) [inline blackbox/docs/configuration.txt]: "used out of" -> "used from"

File: blackbox/docs/configuration.txt [hunk 12]
```diff
@@ -752,15 +769,29 @@ Collecting Stats
...
+  Reaching the limit logs an error message and causes :ref:`sys.jobs_log <sys-logs>` to be
+  cleared.
```
Comment (nslater) [inline blackbox/docs/configuration.txt]: just for my own understanding, by "clears" do you mean completely? or just enough to keep within the memory limits?

File: blackbox/docs/configuration.txt [hunk 13]
```diff
@@ -752,15 +769,29 @@ Collecting Stats
...
+
+**stats.breaker.operations_log.limit**
+  | *Default:*    ``5%``
+  | *Runtime:*   ``yes``
+
+  The operations log circuit breaker prevents that :ref:`sys.operations_log <sys-logs>`
+  exceeds a certain amount of memory by removing old log entries. The value
+  specifies the maximum of memory that can be used out of :ref:`crate-heap-size`.
+  Reaching the limit logs an error message and causes :ref:`sys.operations_log <sys-logs>`
+  to be cleared.
```
Comment (nslater) [inline blackbox/docs/configuration.txt]: same edits as above

File: blackbox/docs/configuration.txt (hunk 12 above)
Comment (mikethebeer) [inline blackbox/docs/configuration.txt]: it will be completely cleared.

File: blackbox/docs/configuration.txt (hunk 1 above)
Comment (chaudum) [inline blackbox/docs/configuration.txt]: Entries are cleared by a periodically triggered job if they are older than the specified expire time.

File: blackbox/docs/configuration.txt [hunk 14]
```diff
@@ -694,42 +694,57 @@ Collecting Stats
...
 
-  This setting specifies the period when jobs are stored in the ``sys.jobs_log``
-  table. It periodically clears entries that are completed and older than the
-  specified time interval. This setting overrides ``jobs_log_size`` and logs a
-  warning message if both settings are set explicitly. If the value is set to
-  ``0`` the setting gets disabled.
+  Entries of :ref:`sys.jobs_log <sys-logs>` are cleared by a periodically job
+  when they are older than the specified expire time. This setting overrides
+  :ref:`stats.jobs_log_size <stats.jobs_log_size>`. If the value is set to ``0``
+  the time based log entry evicition is disabled.
```
Comment (chaudum) [inline blackbox/docs/configuration.txt]: typo: evicition -> eviction

**Comment by mikethebeer** [comment]
this PR is merged. GH seems not to regognize it.

**Pull request closed without merging**